File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        my_dict = {}
        answer = []

        for i, item in enumerate(nums):
            if target - item in my_dict:
                answer.append(i)
                answer.append(my_dict[target-item])
            else:
                my_dict[item] = i

        return answer

File: 1-two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    let my_dict: {[key: string] : number} = {};
    let answer: number[] = [];

    nums.forEach((item, i) => {
        if (my_dict[(target - item)] != undefined){
            answer.push(i);
            answer.push(my_dict[target-item])
        }
        else my_dict[item] = i;
    });
    return answer; 
};